pkg/workspace/apply.go

```diff
@@ -121,3 +140,45 @@ func Apply(ts v1beta1.TaskSpec, wb []v1beta1.WorkspaceBinding, v map[string]core
 	}
 	return &ts, nil
 }
+
+// mountAsSharedWorkspace takes a volumeMount and adds it to all the steps and sidecars in
+// a TaskSpec.
+func mountAsSharedWorkspace(ts v1beta1.TaskSpec, volumeMount corev1.VolumeMount) {
+	ts.StepTemplate.VolumeMounts = append(ts.StepTemplate.VolumeMounts, volumeMount)
+
+	for i := range ts.Sidecars {
+		sidecar := &ts.Sidecars[i]
+		sidecar.VolumeMounts = append(sidecar.VolumeMounts, volumeMount)
+	}
+}
+
+// mountAsIsolatedWorkspace takes a volumeMount and adds it only to the steps and sidecars
+// that have requested access to it.
+func mountAsIsolatedWorkspace(ts v1beta1.TaskSpec, workspaceName string, volumeMount corev1.VolumeMount) {
+	for i := range ts.Steps {
+		step := &ts.Steps[i]
+		for _, workspaceUsage := range step.Workspaces {
+			if workspaceUsage.Name == workspaceName {
+				vm := volumeMount
+				if workspaceUsage.MountPath != "" {
+					vm.MountPath = workspaceUsage.MountPath
+				}
+				step.VolumeMounts = append(step.VolumeMounts, vm)
+				break
+			}
+		}
+	}
+	for i := range ts.Sidecars {
+		sidecar := &ts.Sidecars[i]
+		for _, workspaceUsage := range sidecar.Workspaces {
+			if workspaceUsage.Name == workspaceName {
+				vm := volumeMount
+				if workspaceUsage.MountPath != "" {
+					vm.MountPath = workspaceUsage.MountPath
```

Comment:
it seems like this is the code that handles the mountPath overriding logic, but i dont see it being explicitly included in the tests cases below? (apologies if i just missed it) i could see there being an extra case or two for this?

Reply:
i think the coverage report might be backing this up as well:

![image](https://user-images.githubusercontent.com/432502/106949911-a0786000-66fb-11eb-99e6-833bed7c9acf.png)

Reply:
Unit test added to cover custom mountPath in sidecar and step.